nomos-services/cryptarchia-consensus/src/lib.rs

```diff
@@ -142,23 +146,10 @@ impl<State: CryptarchiaState> Cryptarchia<State> {
         }
     }
 
-    /// Prune the in-memory storage of forks older than or equal to the Last
-    /// Immutable Block (LIB).
-    ///
-    /// The old blocks are removed from the consensus engine and their state
-    /// removed from the ledger state.
-    pub fn prune_old_forks(&mut self) -> impl Iterator<Item = HeaderId> {
-        // We need to iterate once, then return the unconsumed iterator, so we need to
-        // collect first.
-        let lib_depth = self
-            .consensus
-            .tip_branch()
-            .length()
-            .checked_sub(self.consensus.lib_branch().length())
-            .expect("Canonical chain tip heigh must be higher than LIB height.");
-        let old_forks_pruned = self.consensus.prune_forks(lib_depth).collect::<Vec<_>>();
+    /// Remove the ledger states of the pruned blocks.
+    pub fn prune_ledger_states(&mut self, pruned_blocks: &PrunedBlocks<HeaderId>) {
```

Comment:
I'd expand on the comment explain which blocks should be removed and which one shouldn't, and maybe remove the pub vis as this is called only internally anyway